core-blocks/button/edit.js

```diff
@@ -106,22 +113,24 @@ class ButtonEdit extends Component {
 						keepPlaceholderOnFocus
 					/>
 					<InspectorControls>
-						<PanelColor
-							colorValue={ backgroundColor.value }
-							title={ __( 'Background Color' ) }
-							onChange={ setBackgroundColor }
-						/>
-						<PanelColor
-							colorValue={ textColor.value }
-							title={ __( 'Text Color' ) }
-							onChange={ setTextColor }
+						<PanelColorSettings
+							title={ __( 'Color Settings' ) }
+							textColorProps={ {
```

Comment:
The fact that we have special props for background and text makes this component specific to this colors and impossible to use for a border color for example.

What if we change the API a little bit to something like:
```
<PanelColorSettings
	title={ __( 'Color Settings' ) }
	colorZones={ [ {
		label: title={ __( 'Background Color' ) }
		value: backgroundColor.value,
		onChange: setBackgroundColor,
	},	{
		label: title={ __( 'Text Color' ) }
		value: textColor.value,
		onChange: setTextColor,
	},
	{
		label: title={ __( 'The Other Color My block Needs' ) }
		value: otherColor.value,
		onChange: setOtherColor,
	}

] }
>
	<ContrastCheckerWithFallbackStyles
			node={ this.nodeRef }
			textColor={ textColor.value }
			backgroundColor={ backgroundColor.value 
			isLargeText={ true }
	/>
</PanelColorSettings>
```

The component would render, all the childs passed so the contrast checking would be something outside of the scope of the component.